templates/common/_base/units/machine-config-daemon-pull.service

```diff
@@ -9,14 +9,20 @@ contents: |
   ConditionPathExists=/etc/pivot/image-pullspec
   ConditionPathExists=/var/lib/kubelet/config.json
   ConditionPathExists=!/usr/local/bin/machine-config-daemon
+  # Pivot should run after MCD image is pulled
   Before=machine-config-daemon-host.service
+  # Start pulling after network is up
+  After=network-online.target
 
   [Service]
-  # Need oneshot to delay kubelet
-  Type=oneshot
+  # Image is being pulled explicitly so that an empty /usr/local/bin/machine-config-daemon won't be
+  # created in case there was a network problem
+  ExecStartPre=/usr/bin/podman --authfile=/var/lib/kubelet/config.json pull '{{ .Images.setupEtcdEnvKey }}'
```

Comment:
TODO: use proxy env vars (see bootkube.sh implementation) to ensure OKD can be installed behind a proxy